### ISSUE TITLE
Support for unordered arrays?

### ISSUE THREAD

**Issue by alex-fedorov** [waterlink/rspec-json_expectations#7]

**Comment by benoittgt** [comment]
I was ready to open an "issue" for that. I'm having random errors when running test with 2 objects.

Like this

```ruby
describe 'get#index' do
    before { sign_in admin }
    let(:store_id) { store.id }
    let!(:user1) { create :user, organisation: organisation }
    let!(:user2) { create :user, organisation: organisation }
    let(:page) { 0 }
    let(:request) { get :index, store_id: store_id, page: page, format: :json }

    context 'with correct store id' do
      before { request }

      context 'index return' do
        it do
          expect(response.body).to include_json(
            users: [
              {
                id:          user1.id,
                last_name:   user1.last_name,
                group_count: 2
              },
              {
                id:          user2.id,
                last_name:   user2.last_name,
                group_count: 1
              }
            ]
          )
        end
      end
    end
  end
```

Sometimes `user1` is created before sometimes `user2`. It's quite anoying. I thought that using json_expectation should match json without order.

**Comment by benoittgt** [comment]
Since then I now specify the order of my json. It's always specify it. 

But it should be great to have something like [match_array](http://www.rubydoc.info/github/rspec/rspec-expectations/RSpec%2FMatchers%3Amatch_array) or [HashWithIndifferentAccess](http://api.rubyonrails.org/classes/ActiveSupport/HashWithIndifferentAccess.html). Will try to have a look but it's seems over-skilled for my junior skill. :smile:

**Comment by waterlink** [comment]
The public API here seems like that: 

```ruby
.to include_json(
  users: UnorderedArray[
     ...
  ]
)
```

WDYT?

**Comment by benoittgt** [comment]
It should be awesome @waterlink

**Comment by waterlink** [comment]
> Will try to have a look but it's seems over-skilled for my junior skill. :smile:

Actually, I have just implemented that, and while doing so, I felt quite embarrassed for the current state of the code :sweat_smile: 

Probably I will find some time and make a 40-60% re-factor or re-write.

**Comment by benoittgt** [comment]
Thanks a lot. Looks good to me. 

For the refactoring that's a big task.

**Comment by waterlink** [comment]
OK, I have released `1.3.0` with this feature. Docs can be found here: https://relishapp.com/waterlink/rspec-json-expectations/docs/json-expectations/unordered-array-matching-support-for-include-json-matcher
